components/calendar/src/persian.rs

```diff
@@ -0,0 +1,959 @@
+        },
+        DateCase {
+            year: 1147,
+            month: 3,
+            day: 30,
+        },
+        DateCase {
+            year: 1198,
+            month: 5,
+            day: 10,
+        },
+        DateCase {
+            year: 1218,
+            month: 1,
+            day: 7,
+        },
+        DateCase {
+            year: 1282,
+            month: 1,
+            day: 29,
+        },
+        DateCase {
+            year: 1308,
+            month: 6,
+            day: 3,
+        },
+        DateCase {
+            year: 1320,
+            month: 7,
+            day: 7,
+        },
+        DateCase {
+            year: 1322,
+            month: 1,
+            day: 29,
+        },
+        DateCase {
+            year: 1322,
+            month: 7,
+            day: 14,
+        },
+        DateCase {
+            year: 1370,
+            month: 12,
+            day: 27,
+        },
+        DateCase {
+            year: 1374,
+            month: 12,
+            day: 6,
+        },
+        DateCase {
+            year: 1417,
+            month: 8,
+            day: 19,
+        },
+        DateCase {
+            year: 1473,
+            month: 4,
+            day: 28,
+        },
+    ];
+
+    // Persian New Year occuring in March of Gregorian year (g_year) to fixed date
+    fn nowruz(g_year: i32) -> RataDie {
+        let iso_from_fixed: Date<Iso> =
+            Iso::iso_from_fixed(RataDie::new(FIXED_PERSIAN_EPOCH.to_i64_date()));
+        let greg_date_from_fixed: Date<Gregorian> = Date::new_from_iso(iso_from_fixed, Gregorian);
+        let persian_year = g_year - greg_date_from_fixed.year().number + 1;
+        let _year = if persian_year <= 0 {
+            persian_year - 1
+        } else {
+            persian_year
+        };
+        #[allow(clippy::unwrap_used)] // valid day and month
+        Persian::fixed_from_persian_integers(_year, 1, 1).unwrap()
+    }
+
+    fn days_in_provided_year_core(year: i32) -> u32 {
+        #[allow(clippy::unwrap_used)] // valid month and day
+        let fixed_year = Persian::fixed_from_persian_integers(year, 1, 1)
+            .unwrap()
+            .to_i64_date();
+        #[allow(clippy::unwrap_used)] // valid month and day
+        let next_fixed_year = Persian::fixed_from_persian_integers(year + 1, 1, 1)
+            .unwrap()
+            .to_i64_date();
+
+        (next_fixed_year - fixed_year) as u32
+    }
+
+    #[test]
+    fn test_persian_leap_year() {
+        let mut leap_years: [i32; 33] = [0; 33];
+        let expected_values = [
+            false, true, false, false, false, false, false, true, false, true, false, false, true,
+            false, false, true, false, false, false, false, false, false, false, true, false,
+            false, false, false, false, true, false, false, false,
+        ];
+
+        let leap_year_cycle: [i32; 2820] = [
+            474, 475, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 490,
```

Comment:
Convert to Rust range. When you do, have local vars for start and end (ex: `canonical_leap_year_cycle_start`, `..._end`).

Alternatively, you can `cycle_start`, `cycle_end` but ensure the comments explain what exactly that is referring to.